server/controllers/finance/conventions.js

```diff
@@ -0,0 +1,196 @@
+/**
+* The /cash/conventions HTTP API endpoint
+*
+* @module finance/conventions
+*
+* @desc This module is responsible for handling all crud operations relatives
+* to convention payment and define all convention api functions
+*
+* @required q
+* @required lib/db
+* @required lib/guid
+* @required lib/util
+* @required ledger
+* @required journal/primarycash
+*
+* @todo implements functions for the GET method
+* @todo implements functions for the PUT method
+*/
+
+'use strict';
+
+var q                  = require('q'),
+    db                 = require('../../lib/db'),
+    uuid               = require('../../lib/guid'),
+    util               = require('../../lib/util'),
+    ledger             = require('./ledger'),
+    journalConvention  = require('./journal/primarycash').convention;
+
+var getDebitorReceipt  = ledger.debitor_group;
+
+/** create a new convention payment record in the database */
+exports.create = create;
+
+/**
+* POST /cash/conventions
+*
+* @desc This function is responsible for creating a new convention payment record in the database
+* and writing to the posting journal
+*
+* @param {object} PrimaryCashRecord A Primary cash record
+*
+* @example
+* // An example of the object sended in the post request
+* // All object's properties below are required
+* {
+*   uuid            : {uuid},
+*   project_id      : {number},
+*   type            : {char},
+*   date            : {date},
+*   currency_id     : {number},
+*   account_id      : {number},
+*   cost            : {number},
+*   user_id         : {number},
+*   description     : {string},
+*   cash_box_id     : {number},
+*   origin_id       : {number}
+* }
+*/
+function create(req, res, next) {
+  var qData            = req.body,
+      exchangeRate     = {},
+      primaryCashItems = [],
+      overviews        = [];
+
+  var query =
+      'SELECT `debitor_group`.`uuid` FROM `debitor_group` ' +
+      'WHERE `debitor_group`.`account_id`=?';
+
+  db.exec(query, [qData.account_id])
```

Comment:
This should be wrapped in a transaction (`var transaction = db.transaction()`). I'll be implementing the same procedure for my cash module tomorrow, so we can compare our different methods.  The problem with this method is that if one SQL fails, the previous records will not be deleted.  Could you experiment implementing this same procedure with transactions?